hudi-common/src/main/java/org/apache/hudi/metadata/HoodieTableMetadataUtil.java

```diff
@@ -842,7 +842,8 @@ public static HoodieData<HoodieRecord> convertFilesToBloomFilterRecords(HoodieEn
 
     List<Pair<String, List<String>>> partitionToDeletedFilesList = partitionToDeletedFiles.entrySet()
         .stream().map(e -> Pair.of(e.getKey(), e.getValue())).collect(Collectors.toList());
-    int parallelism = Math.max(Math.min(partitionToDeletedFilesList.size(), recordsGenerationParams.getBloomIndexParallelism()), 1);
+    int totalDeleteFileSize = partitionToDeletedFilesList.stream().mapToInt(p -> p.getValue().size()).sum();
+    int parallelism = Math.max(Math.min(totalDeleteFileSize, recordsGenerationParams.getBloomIndexParallelism()), 1);
```

Comment:
The number of entries in the list (`partitionToDeletedFilesList`, `partitionToAppendedFiles`) is the number of table partitions.  Parallelizing the list to RDD with a higher parallelism does not have any impact on the execution in Spark since one executor still gets a pair of all info within a table partition, unless the DAG is rewritten.

Reply:
oh, I got it. I mistakenly think that the later processing logic is the same as the update, I will try fix it.